### ISSUE TITLE
Unsloth playbook: test_unsloth.py trains a broken model (flat loss ~7) — unsloth/Phi-3.5-mini-instruct mirror has a defective config; swap to microsoft/Phi-3.5-mini-instruct

### ISSUE THREAD

**Issue by legolaz8451** [NVIDIA/dgx-spark-playbooks#85]
### Summary

The unsloth playbook's validation script `test_unsloth.py` cannot pass its own success criterion ("training loss decreasing") as written. The failure is not environmental: the hardcoded model repo `unsloth/Phi-3.5-mini-instruct` is a llamafied conversion whose `config.json` carries a wrong `attention_factor: 32` in `rope_scaling`. Every transformers release since 4.44 honors that field verbatim, so the model loads in a degraded state and training goes nowhere. This is distinct from the setup problems already tracked in #14/#19.

### Environment

- DGX Spark (GB10, aarch64), DGX OS 7.5.0, kernel 6.17.0-1026-nvidia
- Container: `nvcr.io/nvidia/pytorch:25.11-py3` (torch 2.10.0a0+b558c986e8.nv25.11, CUDA 13.0, triton 3.5.0)
- Installed per playbook README: `transformers 4.56.2`, `trl 0.26.1`, `datasets 4.3.0`, `peft 0.19.1`, `unsloth 2026.7.3`, `unsloth_zoo 2026.7.3`, `bitsandbytes 0.49.2` (second line with `--no-deps`)

### Symptom (playbook as written)

`python test_unsloth.py` runs to completion but the model never learns:

```text
{'loss': 7.32, 'grad_norm': 3.877e+13, 'learning_rate': 0.0, ...}   # step 1, lr=0 → broken at load time
... loss stays ~5.5–7.4 for all 60 steps (step-1 was 6.7–7.3 across three runs), grad_norm 1e12–1e17
```

Reproduced identically with `load_in_4bit = False` (bitsandbytes exonerated) and under transformers 5.14.1 and 4.56.2 (version exonerated). A 15-line vanilla-transformers probe isolates it to the model repo itself:

| repo | loss on "The capital of France is Paris. Water boils at 100 °C…" | greedy completion of "The capital of France is" |
|---|---:|---|
| `unsloth/Phi-3.5-mini-instruct` | **5.354** | "…the most important thing in the world of" |
| `microsoft/Phi-3.5-mini-instruct` | **1.328** | "…**Paris.** Paris is the" |

### Root cause

The mirror's `config.json` (authored 2024-08-20 against transformers 4.44.1, never revised) sets `rope_scaling.attention_factor: 32`. transformers' `_compute_longrope_parameters` uses a config-specified `attention_factor` verbatim; the correct longrope value is `sqrt(1 + ln(32)/ln(4096)) ≈ 1.19`. The ~27× over-scaling of the rotary tables saturates attention—fluent but non-functional output, untrainable. Details reported to unsloth at https://huggingface.co/unsloth/Phi-3.5-mini-instruct/discussions/2—including verification that the mirror's weights themselves are lossless (loading them with the two fields removed from `rope_scaling` scores loss 1.327 vs the original's 1.328); the config is the entire defect.

### Fix (validated end-to-end on DGX Spark)

Two-line change in `test_unsloth.py`:

```python
model, tokenizer = FastModel.from_pretrained(
    model_name = "microsoft/Phi-3.5-mini-instruct",
    use_exact_model_name = True,  # prevent unsloth remapping back to its mirror
    ...
)
```

Result with the otherwise-unmodified playbook: step-1 loss **2.30** (at lr=0), grad_norm **0.24–0.78** throughout, loss trending 2.3 → ~1.4 over 60 steps, `train_loss 1.73`, 76 s runtime. Banner confirms "Fast Phi3 patching", no remap.

### Suggested playbook changes

1. Swap the hardcoded repo to `microsoft/Phi-3.5-mini-instruct` + `use_exact_model_name = True` (or any model whose repo isn't a stale conversion).
2. Consider asserting on the step-1 loss (e.g. `< 4`) in the validation script so a load-time-broken model fails fast instead of burning 60 steps and passing "it ran".
